lms/djangoapps/grades/subsection_grade.py

```diff
@@ -253,7 +253,28 @@ def update_or_create_model(self, student, score_deleted=False, force_update_subs
         Saves or updates the subsection grade in a persisted model.
         """
         if self._should_persist_per_attempted(score_deleted, force_update_subsections):
-            return PersistentSubsectionGrade.update_or_create_grade(**self._persisted_model_params(student))
+            model = PersistentSubsectionGrade.update_or_create_grade(**self._persisted_model_params(student))
```

Comment:
So the returned `model` will not have the same values as the parameters passed in to `update_or_create_grade`?

Reply:
Correct, because there's possibly a `PersistentSubsectionGradeOverride` object associated with this grade that has to be read from the DB before we can know what the final earned/possible grade is.

Reply:
Ok - but why does the in-memory representation of the grade need to be updated?
It seems counter-intuitive from a functional design standpoint as a consumer of the `PersistentSubsectionGrade.update_or_create_grade` API.

Reply:
Yeah, I know.  This all stems from the `SubsectionGrade` class and its children not being "in charge" of overrides right now - the overrides are all dealt with inside the `PersistentSubsectionGrade` class.  `CreateSubsectinGrade` objects never do any kind of query for persistent grades or overrides during their creation - their grade values are determined only from problem scores.

I don't really know how this can be made cleaner (although I haven't been thinking about it that long).  One quick thing might be to move the `PersistentSubsectionGrade._prepare_params_override()` method in the `CreateSubsectionGrade` class.  That way, you'll at least have the values returned be the same as those that are passed in.  Another thing I thought briefly about was always doing an explicit read of persistent subsection grades first, before doing anything with `CreateSubsectionGrade`.  That way you can have the same data consistency that you have during instantiation of the `ReadSubsectionGrade` class.